packages/ra-ui-materialui/src/input/FileInput.tsx

```diff
@@ -163,13 +166,27 @@ export const FileInput = (props: FileInputProps) => {
             resource={resource}
             isRequired={isRequired}
             color={(isTouched || isSubmitted) && invalid ? 'error' : undefined}
+            sx={{
+                cursor: disabled || readOnly ? 'default' : 'pointer',
+                ...rest.sx,
+            }}
             {...sanitizeInputRestProps(rest)}
         >
             <>
                 <div
                     {...getRootProps({
                         className: FileInputClasses.dropZone,
                         'data-testid': 'dropzone',
+                        style: {
+                            color:
+                                disabled || readOnly
+                                    ? 'rgba(0, 0, 0, 0.38)'
+                                    : inputPropsOptions?.color || 'black',
+                            backgroundColor:
+                                disabled || readOnly
+                                    ? 'rgba(0, 0, 0, 0.12)'
+                                    : inputPropsOptions?.backgroundColor,
+                        },
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏We should use the theme (`import {  useTheme } from '@mui/material/styles';`) to keep visibility even in `darkTheme`. Something like:
```js
style: {
    color:
        disabled || readOnly
            ? theme.palette.text.disabled
            : inputPropsOptions?.color ||
                theme.palette.text.primary,
    backgroundColor:
        disabled || readOnly
            ? theme.palette.background.default
            : inputPropsOptions?.backgroundColor,
},
```